presto-benchmark-runner/src/main/java/com/facebook/presto/benchmark/framework/BenchmarkManager.java

```diff
@@ -0,0 +1,173 @@
+                submit(ImmutableList.copyOf(queries.values()), benchmarkSuiteInfo.getSessionProperties());
+            }
+        }
+        reportProgressUntilFinished();
+    }
+
+    @VisibleForTesting
+    AtomicInteger getQueriesSubmitted()
+    {
+        return queriesSubmitted;
+    }
+
+    @Override
+    public synchronized boolean resubmit(BenchmarkQueryRunner benchmarkQueryRunner, BenchmarkQueryException benchmarkQueryException)
+    {
+        if (!shouldResubmit(benchmarkQueryException)) {
+            return false;
+        }
+
+        String name = benchmarkQueryRunner.getQuery().getName();
+        Integer resubmittedCount = resubmittedCounts.getOrDefault(name, 0);
+        if (resubmittedCount >= benchmarkResubmissionLimit) {
+            log.info("Benchmark query %s failed with %s, resubmission limit exceeded", name, benchmarkQueryException.getErrorCode());
+            return false;
+        }
+
+        queriesSubmitted.addAndGet(1);
+        resubmittedCounts.compute(name, (key, count) -> count == null ? 1 : count + 1);
+        completionService.submit(benchmarkQueryRunner::run);
+        log.info("Benchmark query %s failed with %s, resubmitted (%s/%s)", name, benchmarkQueryException.getErrorCode(), resubmittedCount, benchmarkResubmissionLimit);
+        return true;
+    }
+
+    private void submit(List<BenchmarkQuery> benchmarkQueries, Map<String, String> sessionProperties)
+    {
+        for (BenchmarkQuery benchmarkQuery : benchmarkQueries) {
+            BenchmarkQueryRunner benchmarkQueryRunner = benchmarkQueryRunnerFactory.get(this, benchmarkQuery, sessionProperties);
+            completionService.submit(benchmarkQueryRunner::run);
+        }
+        log.info("Queries submitted: %s", benchmarkQueries.size());
+        this.queriesSubmitted.addAndGet(benchmarkQueries.size());
+    }
+
+    private void reportProgressUntilFinished()
+    {
+        int completed = 0;
+        double lastProgress = 0;
+        Map<EventStatus, Integer> statusCount = new EnumMap<>(EventStatus.class);
+
+        while (completed < queriesSubmitted.get()) {
+            try {
+                Optional<BenchmarkQueryEvent> event = completionService.take().get();
+                completed++;
+                if (event.isPresent()) {
+                    statusCount.compute(EventStatus.valueOf(event.get().getStatus()), (status, count) -> count == null ? 1 : count + 1);
+                    postEvent(event.get());
+                }
+
+                double progress = ((double) completed) / queriesSubmitted.get() * 100;
+                if (progress - lastProgress > 0.5 || completed == queriesSubmitted.get()) {
+                    log.info(
+                            "Progress: %s succeeded, %s failed, %.2f%% done",
+                            statusCount.getOrDefault(SUCCEEDED, 0),
+                            statusCount.getOrDefault(FAILED, 0),
+                            progress);
```

Comment:
Progress reporting should just be # of queries vs. # of total queries. Any query failure should result in termination of the run.

Reply:
As discussed, for some runs we may be okay with having query failures. Hence it will be good to show # of failed queries as well.